src/new/wallet.js

```diff
@@ -2666,6 +2675,168 @@ class HathorWallet extends EventEmitter {
+    const methodArgs = get(blueprintInformation, `public_methods.${method}.args`);
+    if (!methodArgs) {
+      throw new NanoContractTransactionError(`Blueprint does not have method ${method}.`);
+    }
+
+    // Args may come as undefined
+    const argsLen = data.args ? data.args.length : 0;
+    if (argsLen !== methodArgs.length) {
+      throw new NanoContractTransactionError(`Method needs ${methodArgs.length} parameters but data has ${data.args.length}.`);
+    }
+
+    // Here we validate that the arguments sent in the data array of args has
+    // the expected type for each parameter of the blueprint method
+    for (const [index, arg] of methodArgs.entries()) {
+      let typeToCheck = arg.type;
+      if (typeToCheck.startsWith('SignedData')) {
+        // Signed data will always be an hexadecimal with the
+        // signed part and the data itself
+        typeToCheck = 'str';
+      }
+      switch (typeToCheck) {
+        case 'bytes':
+          // Bytes arguments are sent in hexadecimal
+          try {
+            data.args[index] = hexToBuffer(data.args[index]);
+          } catch {
+            // Data sent is not a hex
+            throw new NanoContractTransactionError(`Invalid hexadecimal for argument number ${index + 1}.`);
+          }
+          break;
+        case 'int':
+        case 'float':
+          if (typeof data.args[index] !== 'number') {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+          break;
+        case 'str':
+          if (typeof data.args[index] !== 'string') {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+          break;
+        default:
+          if (arg.type !== typeof data.args[index]) {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+      }
+    }
+
+    // Get private key that will sign the nano contract transaction
+    let derivedKey;
+    try {
+      derivedKey = await this.getHDPrivateKeyFromAddress(address, options);
+    } catch (e) {
+      if (e instanceof AddressError) {
+        throw new NanoContractTransactionError('Address used to sign the transaction does not belong to the wallet.');
+      }
+      throw e;
+    }
+    const privateKey = derivedKey.privateKey;
+
+    // Build and send transaction
+    const builder = new NanoContractTransactionBuilder();
+    builder.setMethod(method);
+    builder.setWallet(this);
+    builder.setBlueprintId(blueprintId);
+    builder.setNcId(method === 'initialize' ? null : data.ncId);
+    builder.setCaller(privateKey);
+    builder.setActions(data.actions);
+    builder.setArgs(data.args);
```

Comment:
suggestion(if-minor): If each set method returns `this` we can chain these calls.

This doesn't change much but I think its easier to read if it were like this:

```typescript
const builder = new NanoContractTransactionBuilder()
                     .setMethod(method)
                     .setWallet(this)
                     .setBlueprintId(blueprintId)
                     .setNcId(method === 'initialize' ? null : data.ncId)
                     .setCaller(privateKey)
                     .setActions(data.actions)
                     .setArgs(data.args);
```

Reply:
Much better

Reply:
Indeed, much better. https://github.com/HathorNetwork/hathor-wallet-lib/pull/604/commits/a47329d5f791a0f7c9650eb5c55ed2d2b26a986c